src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -16,40 +15,56 @@
 import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
 
 public class IntakeSubsystem extends SubsystemBase implements AutoCloseable {
-  public static final double kOuttakeSpeed = -1;
-  public static final double kIntakeSpeed = 0.5;
-  public static final double kRetractPosition = -.5; //TODO measure and find position 
-  public static final double kExtendPosition = .5;
+  public static final double kConeIntakeSpeed = -0.75;
+  public static final double kConeOuttakeSpeed = 0.75;
+  public static final double kCubeIntakeSpeed = 0.55;
+  public static final double kCubeOuttakeSpeed = -0.65;
+
   private CANSparkMax motor;
   private CANSparkMax motor2;
-  private CANSparkMax motor3;
-  private boolean inOutArm = false;
   private static ShuffleboardTab matchTab = Shuffleboard.getTab("Match");
   private static GenericEntry intakeEntry = matchTab.add("Intake Speed", 0.0).getEntry();
+  private static GenericEntry intakeMode = matchTab.add("Intake Mode", "Cone Mode").getEntry();
+
+  private boolean isConeMode;
+
   /** Creates a new IntakeSubsystem. */
   public IntakeSubsystem() {
+    isConeMode = true;
+
     motor = MotorController.constructMotor(MotorConfig.IntakeMotor1);
     motor2 = MotorController.constructMotor(MotorConfig.IntakeMotor2);
-    motor.follow(motor2);
   }
   
   private void spinWheels(double velocity) {
     motor.set(velocity);
-
+    motor2.set(-velocity);
+    intakeEntry.setDouble(velocity);
   }
 
   public void push() {
-    spinWheels(kOuttakeSpeed);
+    spinWheels(isConeMode ? kConeOuttakeSpeed : kCubeOuttakeSpeed);
   }
 
   public void pull() {
-    spinWheels(kIntakeSpeed);
+    spinWheels(isConeMode ? kConeIntakeSpeed : kCubeIntakeSpeed);
+  }
+
+  public void setMode(boolean isConeMode) {
+    this.isConeMode = isConeMode;
```

Comment:
Having to know whether true = Cone, or not is confusing.  

Should we have setConeMode, setCubeMode with no parameters?
Or have a constant setMode(instakeSubsystem.coneMode)?

We can put this as a fix in a trello card, and get a new member to make the code changes and push to develop after release3 is merged, but we should still improve this.

Reply:
Sure, put it in a card
Probably a good newcomer problem

Reply:
Zane is making card